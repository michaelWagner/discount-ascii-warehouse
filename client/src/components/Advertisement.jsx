import React from 'react';

class Advertisement extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <img className="ad" src={"/ad/?r=" + Math.floor(Math.random()*1000)} />
    );
  }
}
export default Advertisement;
