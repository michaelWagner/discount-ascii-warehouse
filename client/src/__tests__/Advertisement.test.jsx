import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Advertisement from '../components/Advertisement';

describe(Advertisement, () => {

  const parentComponentName = "ProductGrid";
  const mockGenerateRandomId = jest.fn(() => (325));
  const component = shallow(
    <Advertisement generateRandomId={mockGenerateRandomId}
                   componentName={parentComponentName}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <Advertisement generateRandomId={mockGenerateRandomId}
                   componentName={parentComponentName}/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('generates a random id', () => {
    expect(mockGenerateRandomId).toBeCalled();
  });

  it('updates the Advertisement\'s state with it\'s randomId', () => {
    mockGenerateRandomId()
    expect(component.state('randomId')).not.toEqual(undefined);
  });
});
