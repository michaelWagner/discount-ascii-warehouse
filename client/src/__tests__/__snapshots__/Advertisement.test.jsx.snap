exports[`class Advertisement extends /* istanbul ignore next */_react2.default.Component {
  constructor(props) {
    super(props);

    this.state = {
      randomId: null };

  }

  componentWillMount() {
    let randomId = this.props.generateRandomId();
    this.setState({ randomId: randomId });
  }

  render() {
    const random = this.state.randomId;

    const advertisement =
    this.props.componentName === 'ProductGrid' ?
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'tr', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 22 }, __self: this },
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'td', /* istanbul ignore next */{ className: 'table-advertisement', colSpan: '6', __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this },
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 24 }, __self: this },
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'img', /* istanbul ignore next */{ key: random, className: 'ad', alt: '', src: "/ad/?r=" + random, __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this })))) :



    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'img', /* istanbul ignore next */{ key: random, className: 'ad', alt: '', src: "/ad/?r=" + random, __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this });


    return advertisement;
  }} renders and matches our snapshot 1`] = `
<tr>
  <td
    className="table-advertisement"
    colSpan="6">
    <div>
      <img
        alt=""
        className="ad"
        src="/ad/?r=325" />
    </div>
  </td>
</tr>
`;
